src/auth/authenticator/oicd/jwt.go

```diff
@@ -0,0 +1,82 @@
+package oicd
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/golang-jwt/jwt"
+)
+
+type JWTChecker struct {
+	certs     []*x509.Certificate
+	parser    *jwt.Parser
+	claimsCfg *ClaimsConfig
+}
+
+func NewJWTChecker(certs []*x509.Certificate, claimsCfg *ClaimsConfig, skipClaimsValidation bool) *JWTChecker {
+	return &JWTChecker{
+		certs:     certs,
+		claimsCfg: claimsCfg,
+		parser: &jwt.Parser{
+			SkipClaimsValidation: skipClaimsValidation,
+		},
+	}
+}
+
+func (checker *JWTChecker) Check(_ context.Context, bearerToken string) (*Claims, error) {
+	if len(checker.certs) == 0 {
+		// If no certificate provided we deactivate authentication
+		return nil, nil
+	}
+
+	// Parse and validate token injected in context
+	token, err := checker.parser.ParseWithClaims(
+		bearerToken,
+		&Claims{cfg: checker.claimsCfg},
+		checker.keyFunc,
+	)
+	if err != nil {
+		return nil, err
+	} else if !token.Valid {
+		return nil, fmt.Errorf("invalid access token")
+	}
+
+	return token.Claims.(*Claims), nil
+}
+
+func (checker *JWTChecker) keyFunc(token *jwt.Token) (interface{}, error) {
+	_, ok := token.Method.(*jwt.SigningMethodRSA)
+	if !ok {
+		return nil, fmt.Errorf("invalid access token signing method")
```

Comment:
same comment on Unauthorized error

Reply:
Checker errors are all converted into `Unauthorized` on the Authenticator (As you see below). What would it be the reason to have different errors returned by the checker?
```
claims, err := a.jwtChecker.Check(req.Context(), token)
	if err != nil {
		return nil, errors.UnauthorizedError(err.Error())
	}
```